### DISCUSSION TITLE
CodeSandbox CI not working?

### DISCUSSION THREAD

**Discussion by mrlubos** [codesandbox/codesandbox-client#4656] [General]
Hi! First of all, thank you for making such an amazing product.

**Expect Result**
Now to my issue. What I am trying to achieve is run CodeSandbox CI on every pull request to my `main` branch. (This might change in the future). An example of the desired result can be seen in the [React's repository](https://github.com/facebook/react/pull/19552).

**Actual Result**
What happens? Well... nothing happens.

**Context**
For more context, I am not trying to build a library but rather a whole application. The objective is to allow anyone review pull requests without setting up the project locally. There's more configuration stuff that's not working as expected right now, but that's for later.

**What I did**
I signed up for CodeSandbox and upgraded to personal Pro account. My repository is private so this was a prerequisite. I created a new workspace for the team (only me so far) and signed up for team Pro waitlist.

I then went to Repositories and imported my Github repository. This took me a while as the wizard doesn't really tell you what's wrong when it fails. For example, initially it said the repository cannot be found, but this was a permission problem with the application (it couldn't read repositories, only user emails). Then it failed with the message `No commit found for SHA: master`. Now this is clearer, but still not super user-friendly. Finally, I ended up successfully importing my project from `https://github.com/org/repo/tree/main`.

At this point I have the project in CodeSandbox and while it still doesn't fully work, at least it builds and I can see the source code. So the last step is making the CI run on every pull request.

The [documentation](https://codesandbox.io/docs/ci#how-do-i-set-this-up?) mentions that

>For most libraries, the only thing you need to do is install this GitHub App.

but that seems to be doing nothing. The installed Github App has access to this specific repository by the way, not the whole organisation.

The next thing I tried doing is adding a `.codesandbox/ci.json` configuration file with the following contents.

```json
{
  "sandboxes": ["new"]
}
```

Unfortunately, still the same result, nothing happens. I've tried using `sandboxes: ['vanilla']` instead as well, but that made no difference.

Can anybody help me understand what's going on and how to fix this issue please? Thank you!

**Comment by lbogdan** [comment]
Hey @mrlubos ,

Currently CodeSandbox CI is only addressed to JavaScript packages / libraries. The way it works is it builds every PR, uploads the build artefacts to our own "registry", it forks the sandboxes specified in the config and replaces the package(s) in `package.json` with the PR built versions, so that you can test the PR changes in one (or more) example sandboxes.

I guess for a full application a "traditional" CI would be more suited.

It would also be helpful for us if you could tell us how you were expecting it to work in the first place.

**Reply by mrlubos** [reply]
Hi @lbogdan, thank you for your response!

What you write makes sense. The most attractive feature for me was that CodeSandbox generates a new, secure URL for every build. I thought of it as Netlify's preview URLs, but with the ability to have online IDE in one package, which sounded very cool!
